File: Android_A04E-VoCongDoan-main/Lab3_happyBirthday/app/src/main/java/com/example/happybrithday/MainActivity.java
package com.example.happybrithday;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import org.w3c.dom.Text;

public class MainActivity extends AppCompatActivity {
    TextView textViewHappyBrithday;
    TextView textViewFromEmma;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textViewHappyBrithday=(TextView)findViewById(R.id.txtHappyBrithday);
        textViewHappyBrithday.setText("HAPPY BIRTHDAY SAM!");

        textViewFromEmma=(TextView)findViewById(R.id.txtFromEmma);
        textViewFromEmma.setText("FROM EMMA");

    }
}

File: Android_A04E-VoCongDoan-main/Lab4_workCall/kotlin-words-call-app/app/src/main/java/com/example/wordsapp/MainActivity.kt
package com.example.wordsapp

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.wordsapp.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var recyclerView: RecyclerView
    private var isLinearLayoutManager = true

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)


        val binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        recyclerView = binding.recyclerView
        chooseLayout()
//
    }

    private fun chooseLayout() {
        if (isLinearLayoutManager) {
            recyclerView.layoutManager = LinearLayoutManager(this)
        } else {
            recyclerView.layoutManager = GridLayoutManager(this, 4)
        }
        recyclerView.adapter = LetterAdapterActivity()
    }

    private fun setIcon(menuItem: MenuItem?) {
        if (menuItem == null)
            return
        menuItem.icon =
            if (isLinearLayoutManager)
                ContextCompat.getDrawable(this, R.drawable.ic_grid_layout)
            else ContextCompat.getDrawable(this, R.drawable.ic_linear_layout)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.layout_menu, menu)

        val layoutButton = menu?.findItem(R.id.action_switch_layout)
        setIcon(layoutButton)

        return true
    }


    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.action_switch_layout -> {
                isLinearLayoutManager = !isLinearLayoutManager
                chooseLayout()
                setIcon(item)
                return true
            }
            R.id.action_layout_make_call -> {
//                Toast.makeText(this, "succeess", Toast.LENGTH_LONG).show()
                val intent = Intent(this@MainActivity,com.example.wordsapp.MakeCallActivity::class.java)
                this@MainActivity.startActivity(intent)
                return true
            }

            else -> super.onOptionsItemSelected(item)
        }
    }
}
